src/core_plugins/kibana/public/visualize/listing/visualize_listing.html

```diff
@@ -0,0 +1,156 @@
+        </a>
+      </div>
+    </div>
+
+    <!-- NoResults -->
+    <div class="kuiPanel kuiPanel--centered" ng-if="!listingController.items.length">
+      <div class="kuiNoResults">
+        No visualizations matched your search.
+      </div>
+    </div>
+
+    <!-- Table -->
+    <table class="kuiTable" ng-if="listingController.items.length">
+      <thead>
+        <tr>
+          <th class="kuiTableHeaderCell kuiTableHeaderCell--checkBox">
+            <input
+              type="checkbox"
+              class="kuiCheckBox"
+              ng-checked="listingController.areAllItemsChecked()"
+              ng-click="listingController.toggleAll()"
+            >
+          </th>
+
+          <th class="kuiTableHeaderCell kuiTableHeaderCell--icon">
+          </th>
+
+          <th class="kuiTableHeaderCell">
+            Name
+          </th>
+
+          <th class="kuiTableHeaderCell">
+            Type
+          </th>
+        </tr>
+      </thead>
+
+      <tbody>
+        <tr
+          ng-repeat="item in listingController.items track by item.id | orderBy:'title'"
+          class="kuiTableRow"
+        >
+          <td class="kuiTableRowCell kuiTableRowCell--checkBox">
+            <input
+              type="checkbox"
+              class="kuiCheckBox"
+              ng-click="listingController.toggleItem(item)"
+              ng-checked="listingController.isItemChecked(item)"
+            >
+          </td>
+
+          <td class="kuiTableRowCell kuiTableRowCell--icon">
+            <div class="kuiIcon kuiIcon--basic {{ item.icon }}"></div>
+          </td>
+
+          <td class="kuiTableRowCell">
+            <div class="kuiTableRowCell__liner">
+              <a class="kuiLink" ng-href="{{ item.url }}">
+                {{ item.title }}
+              </a>
+
+              <button
+                class="kuiMicroButton kuiTableRowHoverReveal"
+                ng-click="listingController.edit(item)"
+                aria-label="Edit"
+                tooltip="Edit object"
+              >
+                <span
+                  aria-hidden="true"
+                  class="kuiIcon fa-code"
+                ></span>
+              </button>
+            </div>
+          </td>
+
+          <td class="kuiTableRowCell">
+            <div class="kuiTableRowCell__liner">
+              {{ item.type.title }}
+            </div>
+          </td>
+        </tr>
+
+      </tbody>
+    </table>
+
+    <!-- ToolBarFooter -->
+    <div class="kuiToolBarFooter">
+      <div class="kuiToolBarFooterSection">
+        <div class="kuiToolBarText" ng-hide="listingController.getSelectedItemsCount() === 0">
+          {{ listingController.getSelectedItemsCount() }} selected
+        </div>
+      </div>
+      <div class="kuiToolBarFooterSection">
+        <!-- We need an empty section for the buttons to be positioned consistently. -->
+      </div>
+    </div>
+  </div>
+</div>
```

Comment:
The previous table component had max 20 items shown at one time.  Will this show all of them? Will the page be able to handle many visualizations?  It's not paged, right, they are just all listed?  I wonder how many visualizations some of our biggest users have, order of 1,000s maybe?